src/kakarot/gas.cairo

```diff
@@ -182,4 +182,20 @@ namespace Gas {
         }
         return max_allowed.low;
     }
+
+    // @notice Compute the access list cost.
+    // @param access_list_len The length of the access list
+    // @param access_list The access list
+    func compute_access_list_gas(access_list_len: felt, access_list: felt*) -> felt {
```

Comment:
this was inlined in `_cache_access_list` because we can cache the access list AND compute the gas in the same loop instead of doing two different O(n) operations

Reply:
Indeed, it was discussed with @ClementWalter. The idea for this function is to move the gas validation outside of the interpreter and do it in the `execute_from_outside` allowing to avoid the execution if the gas validation fails.